crates/router/src/core/payments/flows/session_flow.rs

```diff
@@ -212,17 +193,67 @@ async fn create_applepay_session_token(
             .merchant_identifier,
     };
 
-    let response_router_data = types::PaymentsSessionRouterData {
-        response: Ok(types::PaymentsResponseData::SessionResponse {
-            session_token: payment_types::SessionToken::ApplePay(Box::new(
-                payment_types::ApplepaySessionTokenResponse {
-                    session_token_data: session_response,
-                    payment_request_data: applepay_payment_request,
-                    connector: connector.connector_name.to_string(),
-                },
-            )),
-        }),
-        ..router_data.clone()
+    let delayed_response = delayed_response.contains(&connector_name);
+
+    let response_router_data = if delayed_response {
+        types::PaymentsSessionRouterData {
+            response: Ok(types::PaymentsResponseData::SessionResponse {
+                session_token: payment_types::SessionToken::ApplePay(Box::new(
+                    payment_types::ApplepaySessionTokenResponse {
+                        session_token_data:
+                            payment_types::ApplePaySessionResponse::NoSessionResponse,
+                        payment_request_data: None,
+                        connector: connector_name.to_string(),
+                        delayed_response,
+                    },
+                )),
+            }),
+            ..router_data.clone()
+        }
+    } else {
+        let applepay_session_request = mk_applepay_session_request(state, router_data)?;
+        let response = services::call_connector_api(state, applepay_session_request)
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Failure in calling connector api")?;
+        let session_response: payment_types::NoThirdPartySDKSessionResponse = match response {
+            Ok(resp) => resp
+                .response
+                .parse_struct("NoThirdPartySDKSessionResponse")
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to parse ApplePaySessionResponse struct"),
+            Err(err) => {
+                let error_response: payment_types::ApplepayErrorResponse = err
+                    .response
+                    .parse_struct("ApplepayErrorResponse")
+                    .change_context(errors::ApiErrorResponse::InternalServerError)
+                    .attach_printable("Failed to parse ApplepayErrorResponse struct")?;
+                Err(
```

Comment:
If we got a Error response then why we need to throw that as InternalServerError.

Reply:
why we need to throw a InternalServerError? The reason for throwing the error as InternalServerError?